src/main/java/service/GitLabService.java

```diff
@@ -187,11 +187,14 @@ private void mergePreviousAutoMr(CascadeResult result, String gitlabEventUUID, L
 		getOpenMergeRequests(gitlabEventUUID, projectId)
 				.filter(openMr -> openMr.getTargetBranch().equals(sourceBranch) &&
 						openMr.getSourceBranch().matches(branchPattern))
-				.sorted(Comparator.comparingLong(MergeRequest::getIid))
-				.findFirst()
+				.min(Comparator.comparingLong(MergeRequest::getIid))
 				.ifPresent(mr -> {
 					Log.infof("GitlabEvent: '%s' | Found auto merge request between '%s' and '%s' - iid: '%s', mergeWhenPipelineSucceeds: '%s'", gitlabEventUUID, prevSourceBranch, sourceBranch, mr.getIid(), mr.getMergeWhenPipelineSucceeds());
 					if (mr.getMergeWhenPipelineSucceeds() == Boolean.FALSE) {
+						if (mr.getHasConflicts() != null && mr.getHasConflicts()) {
+							sleepSeconds(5);
```

Comment:
I would add a comment to explain why we need to wait:
```suggestion
							//wait 5 seconds in order for GitLab to have time to update the merge request status
							sleepSeconds(5);
```